float64[] rgb
int64 cam_pos
